cranelift/codegen/src/machinst/abi.rs

```diff
@@ -2464,11 +2464,25 @@ impl<T> CallInfo<T> {
                         // returns), we use the integer temp register in
                         // steps.
                         let parts = (ty.bytes() + M::word_bytes() - 1) / M::word_bytes();
+                        let load_ty = if parts > 1 {
+                            M::word_type()
+                        } else {
+                            if ty.bytes() < M::word_bytes() {
+                                match ty.bytes() {
+                                    1 => I8,
+                                    2 => I16,
+                                    4 => I32,
+                                    _ => unreachable!(),
+                                }
+                            } else {
+                                M::word_type()
+                            }
```

Comment:
This whole expression could be wrapped up in a `Type::int_for_size(bytes: usize)` helper, probably? (And IMHO the if-else split is a little clearer as something like `Type::int_for_size(cmp::min(M::word_bytes(), ty.bytes()))`; maybe together with renaming `load_ty` to `one_part_load_ty` to make it clear why we clamp it)